File: all-submissions/0874-backspace-string-compare/solution.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        StringBuilder sb = new StringBuilder();
        for (char c: s.toCharArray()){
            if(c != '#') {
                sb.append(c);
            } else {
                if(sb.length() > 0){
                    sb.deleteCharAt(sb.length() - 1);
                }
            }

        }
        StringBuilder tb = new StringBuilder();
        for (char c: t.toCharArray()){
            if(c != '#') {
                tb.append(c);
            } else {
                if(tb.length() > 0){
                    tb.deleteCharAt(tb.length() - 1);
                }
            }

        }
        String newS = sb.toString();
        String newT = tb.toString();
        return newS.equals(newT);
        
        
    }
}

File: all-submissions/0874-backspace-string-compare/solution.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        stack_s = []
        stack_t = []
        for ch in s:
            if stack_s:
                if ch == "#":
                    stack_s.pop()
                else:
                    stack_s.append(ch)
            else:
                if ch != "#":
                    stack_s.append(ch)
        for ch in t:
            if stack_t:
                if ch == "#":
                    stack_t.pop()
                else:
                    stack_t.append(ch)
            else:
                if ch != "#":
                    stack_t.append(ch)
        return stack_s == stack_t
